tests/data/CMakeLists.txt

```diff
@@ -27,7 +27,7 @@ set(BTF_DATA_FILE ${CMAKE_CURRENT_BINARY_DIR}/btf_data)
 add_custom_command(
   OUTPUT ${BTF_DATA_FILE}
   COMMAND ${LLVM_OBJCOPY} --dump-section .BTF=${BTF_DATA_FILE} ${DATA_SOURCE_O}
-  DEPENDS data_source_o)
+  DEPENDS data_source_o ${DATA_SOURCE_O})
```

Comment:
Is it possible to put both these values into a variable? 1) to reduce duplication. 2) so you can add a comment describing the issue it's fixing. 

great find for something extremely non-obvious. I recall I tried to fix this when @ryantimwilson hit this a while ago. Didn't manage to find it.

Reply:
Yeah, good idea, I added a new variable.

I'm still amazed that this behaviour is actually *by design* of CMake.